fix: Do not rely on visitFrame calls

romtsn opened the pull request:
## :scroll: Description
<!--- Describe your changes in detail -->
* Do not rely on `visitFrame` calls during instrumentation, as they can be skipped depending on flag set to the AGP (the flag can be overridden by any plugin doing bytecode transformation)
* Adapted VisitorTest to reflect AGP behavior
* Added integration test with firebase-perf plugin

## :bulb: Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here. -->
Closes #228

## :green_heart: How did you test it?


## :pencil: Checklist
<!--- Put an `x` in the boxes that apply -->
- [x] I reviewed the submitted code
- [x] I added tests to verify the changes
- [ ] I updated the docs if needed
- [x] No breaking changes


## :crystal_ball: Next steps

marandaneto commented on CHANGELOG.md:
```diff
@@ -3,6 +3,7 @@
 ## Unreleased
 
 * Bump sentry-cli 1.72.0 which prevent daemonize mode from crashing upload process (#262)
+* Fix: Incompatibilities with other Gradle plugins using the same API from AGP for bytecode instrumentation (#228)
```
```suggestion
* Fix: Incompatibilities with other Gradle plugins using the same API from AGP for bytecode instrumentation (#270)
```

marandaneto reviewed (approved):
I cannot validate if the fix works, but if CI is happy, I am happy, codewise is also good to go, thanks for figuring it out.

romtsn commented:
@marandaneto yep, I added an integration test with firebase-perf plugin, if I revert this fix, the. test would fail (also, the VisitorTest would fail, after I changed the flag to `SKIP_FRAMES`)

marandaneto commented on .github/workflows/integration-tests.yml:
```diff
@@ -59,62 +59,63 @@ jobs:
-        with:
-          name: test-reports
-          path: mobile/build/reports/
+# https://github.com/getsentry/sentry-android-gradle-plugin/issues/271 we'll switch to testing the duckduckgo app
```
Can we make a separate PR for that? so we merge into the main branch and merge it back to this PR.
